runtime/compiler/control/DLLMain.cpp

```diff
@@ -562,16 +562,24 @@ IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void * reserved)
             if (isAOT && !isJIT && TR::Options::getAOTCmdLineOptions()->getOption(TR_NoStoreAOT))
                  TR::Options::getCmdLineOptions()->setOption(TR_DisableInterpreterProfiling, true);
 
+            // If -Xnojit, then recompilation is not supported
+            if (!TR::Options::canJITCompile())
+               {
+               TR::Options::getJITCmdLineOptions()->setAllowRecompilation(false);
+               TR::Options::getAOTCmdLineOptions()->setAllowRecompilation(false);
+               }
 
-            if (!TR::Options::canJITCompile()) // -Xnojit, then recompilation is not supported
-                                              // Ensure JIT and AOT flags are set appropriately
-              {
-              TR::Options::getAOTCmdLineOptions()->setAllowRecompilation(false);
-              TR::Options::getAOTCmdLineOptions()->setOption(TR_DisableCHOpts);
-
-              TR::Options::getJITCmdLineOptions()->setAllowRecompilation(false);
-              TR::Options::getJITCmdLineOptions()->setOption(TR_DisableCHOpts);
-              }
+            // if AOT is not possible and recompilation is disabled,
+            // disable CH Table opts
+            if (!TR::Options::sharedClassCache())
```

Comment:
I don't fully understand this code. 
If SCC/AOT is completely disabled, then: `noRecomp ==> no CHOpts`. Let's say there is  a reason for this implication. 
What happens in the "default" case where both JIT and AOT compilations are allowed. Why don't we enforce the same policy `noRecomp ==> no CHOpts`? It's possible to run with `-Xjit:noRecompile` and whatever breaks if we don't disable CHOpts when recompilation is disabled, is going to break in Xjit Xaot case too.

Reply:
Yeah part of the reason I changed this PR back to a draft is because I realized the logic isn't accurate, so I need to ensure that we don't break existing behaviour (ie, behaviour when JIT compilation is possible). I'll un-draft once I also have a testcase, which is a little non-trivial to write up because it needs multiple stages.